File: frontend/src/types/revenue.ts
// Type rieng cho module Doanh thu - khong dung chung voi types/models.ts (HotelDashboard...)
// vi domain revenue tach doc lap hoan toan khoi domain dashboard.

export interface DailyRevenuePoint {
  date: string;
  revenue: number;
}

export interface RevenueByRoomTypeItem {
  room_type_id: number;
  room_type_name: string;
  revenue: number;
}

export interface HotelRevenueDashboard {
  hotel_id: number;
  from_date: string;
  to_date: string;
  revenue: number;
  revenue_change_pct: number | null;
  occupancy_rate: number;
  occupancy_rate_change_pct: number | null;
  total_bookings: number;
  total_bookings_change_pct: number | null;
  adr: number;
  adr_change_pct: number | null;
  room_revenue: number;
  service_revenue: number;
  daily_revenue: DailyRevenuePoint[];
  top_room_types: RevenueByRoomTypeItem[];
}

File: backend/app/schemas/revenue.py
from datetime import date

from pydantic import BaseModel


# Schema 1 diem doanh thu theo ngay (dung ve bieu do xu huong).
class DailyRevenuePoint(BaseModel):
    date: date
    revenue: float


# Schema 1 dong trong bang xep hang loai phong theo doanh thu.
class RevenueByRoomTypeItem(BaseModel):
    room_type_id: int
    room_type_name: str
    revenue: float


# Schema du lieu tra ve trang Doanh thu cua 1 khach san.
class HotelRevenueDashboardResponse(BaseModel):
    hotel_id: int
    from_date: date
    to_date: date
    revenue: float
    revenue_change_pct: float | None = None
    occupancy_rate: float
    occupancy_rate_change_pct: float | None = None
    total_bookings: int
    total_bookings_change_pct: float | None = None
    adr: float
    adr_change_pct: float | None = None
    room_revenue: float
    service_revenue: float
    daily_revenue: list[DailyRevenuePoint]
    top_room_types: list[RevenueByRoomTypeItem]
